### PR TITLE
Adding BuilderOptions.toBuilder() to allow further configuration.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java [hunk 1]
```diff
@@ -72,6 +72,24 @@
...
+      this.userAgent = original.userAgent;
+      this.dataHost = original.dataHost;
+      this.tableAdminHost = original.tableAdminHost;
+      this.clusterAdminHost = original.clusterAdminHost;
+      this.port = original.port;
+      this.overrideIp = original.overrideIp;
+      this.credentialOptions = original.credentialOptions;
+      this.retryOptions = original.retryOptions;
+      this.timeoutMs = original.timeoutMs;
+    }
```
Comment (gamorris) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: dataChannelCount?

Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: Good catch.  I added it.  I also noticed a discrepancy between the builder and the BigtableOptions naming convention (dataChannelCount vs. channelCount).

**Comment by sduskis** [comment]
Updated.  PTAL.

**Comment by gamorris** [comment]
lgtm

**Pull request merged**